include/quicr/message_buffer.h

```diff
@@ -1,74 +1,127 @@
 #pragma once
 
-#include <quicr/quicr_name.h>
-#include <quicr/quicr_namespace.h>
-
+#include <cassert>
+#include <ostream>
 #include <vector>
 
 namespace quicr {
 /**
  * @brief Variable length integer
  */
-enum class uintVar_t : uint64_t
+class uintVar_t
 {
+public:
+  uintVar_t() = default;
+  uintVar_t(const uintVar_t&) = default;
+  constexpr uintVar_t(uint64_t v)
+    : _value{ v }
+  {
+    assert(v < 0x1ull << 61);
+  }
+
+  constexpr operator uint64_t() const { return _value; }
+  constexpr uintVar_t& operator=(uintVar_t other)
+  {
+    _value = other._value;
+    return *this;
+  }
+  constexpr uintVar_t& operator=(uint64_t value)
+  {
+    assert(value < 0x1ull << 61);
+    _value = value;
+    return *this;
+  }
+
+  friend constexpr bool operator==(uintVar_t a, uintVar_t b)
+  {
+    return a._value == b._value;
+  }
+  friend constexpr bool operator!=(uintVar_t a, uintVar_t b)
+  {
+    return !(a == b);
+  }
+  friend constexpr bool operator>(uintVar_t a, uintVar_t b)
+  {
+    return a._value > b._value;
+  }
+  friend constexpr bool operator>=(uintVar_t a, uintVar_t b)
+  {
+    return a._value >= b._value;
+  }
+  friend constexpr bool operator<(uintVar_t a, uintVar_t b)
+  {
+    return a._value < b._value;
+  }
+  friend constexpr bool operator<=(uintVar_t a, uintVar_t b)
+  {
+    return a._value <= b._value;
+  }
+
+  friend std::ostream& operator<<(std::ostream& os, uintVar_t v)
+  {
+    return os << v._value;
+  }
+
+private:
+  uint64_t _value;
 };
 
-uintVar_t to_varint(uint64_t);
-uint64_t from_varint(uintVar_t);
+[[deprecated("uint64_t is explicitly convertible to uintVar_t. Use assignment "
```

Comment:
Just for readability, it seems like this deprecation warning ought to be on a line before the function declaration.  Not a big deal, so feel free to ignore it.